### PR TITLE
Removed old view controllers, added seedManager wrapper for handling …

### PR DESCRIPTION
**Pull request by ryansheh**
…generating and deriving keys from Nova + simple test, consolidated old keychain/auth logic into AccountKeychain, wrapper for pin and keys in keychain

### CODE REVIEW

File: DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift [hunk 1]
```diff
@@ -36,8 +45,82 @@ class AccountKeychain {
         return isAuthorized
     }
     
-    public func clearAuthorization() {
+    public func clearAuthorization() throws {
+        try? deleteKey()
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift]: nit: surround in a `do` `catch` and throw specific error when we fail to delete the key

File: DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift [hunk 2]
```diff
@@ -36,8 +45,82 @@ class AccountKeychain {
...
+            }
+        }
+    }
+
+    private func load(withKey key: String) -> String? {
+        let query = keychainQuery(withKey: key)
+        query.setValue(kCFBooleanTrue, forKey: kSecReturnData as String)
+        query.setValue(kCFBooleanTrue, forKey: kSecReturnAttributes as String)
+        var result: CFTypeRef?
+        let status = SecItemCopyMatching(query, &result)
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift:98]: Why are we comparing results here?

File: DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift [hunk 3]
```diff
@@ -36,8 +45,82 @@ class AccountKeychain {
...
+            let resultsData = resultsDict.value(forKey: kSecValueData as String) as? Data,
+            status == noErr
+        else {
+            return nil
+        }
+        return String(data: resultsData, encoding: .utf8)
+    }
+
+    private func keychainQuery(withKey key: String) -> NSMutableDictionary {
+        let result = NSMutableDictionary()
```
Comment (BenJamminnn) [inline DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift:110]: nit: if this this a `[String: String]` we can init it as
`let result = [String: String]()` for slightly better readability in regards to intent

**Comment by BenJamminnn** [comment]
Where do we retrieve the mnemonic from? `AccountKeychain`?

File: DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift (hunk 2 above)
Comment (ryansheh) [reply DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift:98]: It's not a comparison but a strange way of querying and storing results per 

"It searches the keychain for items that match the specified query criteria.
If matching items are found, it retrieves the data associated with those items.
It returns the retrieved data (if any) in the result parameter."

File: DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift (hunk 3 above)
Comment (ryansheh) [reply DSNP-Wallet/Source/Common/Auth/AccountKeychain.swift:110]: Good suggestion, but I'd like to stick to NSMutableDictionary bc Keychain Services API is in Obj-C and I think it makes interacting with it in this instance easier.

**Review by BenJamminnn — APPROVED** [review]

**Pull request merged**